internal/internal_test.go

```diff
@@ -622,6 +622,65 @@ func TestCheckContextTimeout(t *testing.T) {
 	assertErrorCounterMetric(t, server, CheckRequestTimeoutErr)
 }
 
+type testServer struct {
+	*envoyExtAuthzGrpcServer
+	beforeCheck func()
+}
+
+func (s *testServer) Check(ctx context.Context, req *ext_authz.CheckRequest) (*ext_authz.CheckResponse, error) {
+	if s.beforeCheck != nil {
+		s.beforeCheck()
+	}
+	return s.envoyExtAuthzGrpcServer.Check(ctx, req)
+}
+
+func TestCheckContextCancelled(t *testing.T) {
+	var req ext_authz.CheckRequest
+	if err := util.Unmarshal([]byte(exampleAllowedRequest), &req); err != nil {
+		panic(err)
```

Comment:
We should use `t.Error` here to convey an error message for the test failure, rather than just panic. I'm not sure you need to stop execution of other tests anyways, which `panic()/t.Fail()` will do.

Reply:
i just kept it similar to other tests 
```
if err := util.Unmarshal([]byte(exampleDeniedRequest), &req); err != nil {
		panic(err)
	}
```